app/javascript/react/components/ReviewForm.js

```diff
@@ -0,0 +1,110 @@
+import { post } from 'fetch-mock'
+import React, { useEffect, useState } from 'react'
+
+const ReviewForm = (props) => {
+  const [getReviewData, setReviewData] = useState([])
+  const [getNewReview, setNewReview] = useState({masks_customers: false, masks_employees: false, social_distancing: false, party_size: false })
+  const [errorList, setErrorList] = useState([])
+
+  const handleInputChange = event => {
+
+    setNewReview({
+      ...getNewReview,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  let handleSubmit = (event) => {
+    
+    event.preventDefault()
+    let formPayload = {
+      review: getNewReview
+    }
+    fetch(`/api/v1/restaurants/${props.id}/reviews`, {
+      credentials: "same-origin",
+      method: "POST",
+      body: JSON.stringify(formPayload),
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      }
+  })
+  .then(response => {
+    if (response.ok) {
+      return response
+  } else {
+      let errorMessage = `${response.status} (${response.statusText})`,
+        error = new Error(errorMessage)
+      throw error
+    }
+  })
+  .then(response => response.json())
+  .then(responseBody => {
+    if(responseBody.review) {
+      setReviewData([...getReviewData, getNewReview])
```

Comment:
We can see in the return that this component is not responsible for displaying the reviews, so when a new review is added we should be adding it to where that information is stored in state. This ReviewForm component should only be responsible for maintaining state for the form and its errors. The parent component holds state for the restaurant the review is for, as well as its reviews. We should be updating state in the parent component when submitting a new review so that the review list is updated accordingly with the new data.

Reply:
The changes regarding the ReviewForm looks to be split between this branch and Sams PR #18. This branch has the state for the review form in the correct component, but PR #18 has the fetch request in the correct component.